Spatial.gd

```diff
@@ -833,28 +854,28 @@ func _on_NodeEditorDialog_hide():
 
 func _on_DeleteNode_pressed():
 	if self.currently_selected_node.point_type == "icon":
-		self.currently_selected_node.object_link.get_parent().remove_child(self.currently_selected_node.object_link)
-		self.currently_selected_node.object_link.queue_free()
+		self.currently_selected_node.icon.get_parent().remove_child(self.currently_selected_node.icon)
+		self.currently_selected_node.icon.queue_free()
 	elif self.currently_selected_node.point_type == "path":
-		var path =   self.currently_selected_node.object_link
-		var path2d = self.currently_selected_node.object_2d_link
-		var index =  self.currently_selected_node.object_index
+		var path =   self.currently_selected_node.path
+		var path2d = self.currently_selected_node.path2d
+		var index =  self.currently_selected_node.point_index
 
 		path.remove_point(index)
 		path2d.remove_point(index)
 	clear_adjustment_nodes()
 	gen_adjustment_nodes()
-	on_gizmo_deselected(self.currently_selected_node)
+	on_gizmo_deselected(self.currently_selected_node.gizmo)
 
 
 func _on_NewNodeAfter_pressed():
 	if self.currently_selected_node.point_type == "icon":
```

Comment:
Can we have logic that does not rely on `point_type` branching in multiple places?